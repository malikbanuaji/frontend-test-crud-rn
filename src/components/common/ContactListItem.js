import React from 'react';
import {StyleSheet, Text, TouchableNativeFeedback, View} from 'react-native';
import {Spacing, Typography} from '../../styles';
import PhotoSmall from './PhotoSmall';

export default function ContactListItem({
  id,
  photo,
  firstName,
  lastName,
  age,
  onPress,
  style,
}) {
  function _handlePress() {
    onPress && onPress({id: id});
  }

  return (
    <View style={[styles.contactListItem, style]}>
      <TouchableNativeFeedback
        testID={'contact-list-item-button'}
        onPress={_handlePress}>
        <View style={styles.contactListItemButton}>
          <PhotoSmall
            style={styles.photoSmall}
            photo={photo}
            firstName={firstName}
            lastName={lastName}
          />
          <View style={styles.contentContainer}>
            <Text style={styles.name}>{`${firstName} ${lastName}`}</Text>
            <Text style={styles.age}>{age}</Text>
          </View>
        </View>
      </TouchableNativeFeedback>
    </View>
  );
}

const styles = StyleSheet.create({
  contactListItemButton: {
    alignItems: 'center',
    paddingHorizontal: Spacing.base,
    paddingVertical: Spacing.smaller,
    display: 'flex',
    flexDirection: 'row',
  },
  contentContainer: {
    display: 'flex',
    flexDirection: 'column',
  },
  name: {
    ...Typography.cardTitle,
  },
  age: {
    ...Typography.cardSubtitle,
  },
  photoSmall: {
    marginRight: Spacing.smaller,
  },
});
