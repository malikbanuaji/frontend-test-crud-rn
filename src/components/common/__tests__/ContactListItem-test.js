import React from 'react';

import renderer from 'react-test-renderer';
import ContactListItem from '../ContactListItem';

test('renders correctly', () => {
  const tree = renderer.create(<ContactListItem />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly without image', () => {
  const tree = renderer.create(<ContactListItem photo={'N/A'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders correctly only age', () => {
  const tree = renderer.create(<ContactListItem age={'23'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly only lastName', () => {
  const tree = renderer.create(<ContactListItem lastName={'Doe'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly only firstname', () => {
  const tree = renderer.create(<ContactListItem firstName={'John'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with full props', () => {
  const tree = renderer
    .create(
      <ContactListItem
        firstName={'John'}
        lastName={'Doe'}
        photo={
          'https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png'
        }
        age={23}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
