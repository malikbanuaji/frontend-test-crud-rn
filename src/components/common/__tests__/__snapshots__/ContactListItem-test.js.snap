// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#93b5e1",
              "borderRadius": 8,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
          }
        }
      >
        undefined undefined
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Light",
            "fontSize": 14,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly only age 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#93b5e1",
              "borderRadius": 8,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
          }
        }
      >
        undefined undefined
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Light",
            "fontSize": 14,
          }
        }
      >
        23
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly only firstname 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#93b5e1",
              "borderRadius": 8,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
            }
          }
        >
          J
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
          }
        }
      >
        John undefined
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Light",
            "fontSize": 14,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly only lastName 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#93b5e1",
              "borderRadius": 8,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
            }
          }
        >
          D
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
          }
        }
      >
        undefined Doe
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Light",
            "fontSize": 14,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly with full props 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 8,
          },
        ]
      }
    >
      <Image
        height={40}
        source={
          Object {
            "uri": "https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png",
          }
        }
        style={
          Object {
            "borderRadius": 8,
            "height": 40,
            "overflow": "hidden",
            "width": 40,
          }
        }
        width={40}
      />
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
          }
        }
      >
        John Doe
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Light",
            "fontSize": 14,
          }
        }
      >
        23
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly without image 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#93b5e1",
              "borderRadius": 8,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
          }
        }
      >
        undefined undefined
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Light",
            "fontSize": 14,
          }
        }
      />
    </View>
  </View>
</View>
`;
